### PR TITLE
fix unary expressions.

### PR DESCRIPTION
**Pull request by yu-shipit**
1. support queries with multiple unary sign such as -+--foo
 2. fix rhs scalar unary expression. Basically, the right plan for this case should be `ScalarBinaryOperationExec` instead of `BinaryJoinExec`.

**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** (link exiting issues here : https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests)
1. do not handle RHS unary expression properly.
2. do not support multiple unary expressions.

**New behavior :**
can handle queries like foo > -1
support queries with multiple unary signs like ---foo > -+-1

**BREAKING CHANGES**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

Breaking changes may include:
- Any schema changes to any Cassandra tables
- The serialized format for Dataset and Column (see .toString methods)
- Over the wire formats for Akka messages / case classes
- Changes to the HTTP public API
- Changes to query parsing / PromQL parsing

**Other information**:

### CODE REVIEW

File: coordinator/src/test/scala/filodb.coordinator/queryplanner/PlannerHierarchySpec.scala [hunk 1]
```diff
@@ -182,6 +182,140 @@ class PlannerHierarchySpec extends AnyFunSpec with Matchers with PlanValidationS
 
   private val queryParams = PromQlQueryParams("notUsedQuery", 100, 1, 1000)
 
+  it("Plan with unary expression should be equals to its binary counterpart.") {
```
Comment (alextheimer) [inline coordinator/src/test/scala/filodb.coordinator/queryplanner/PlannerHierarchySpec.scala:185]: Can you add one test in this file for something like this?
```
-(foo{} + bar{})
```
Just to triple-check that a root-level unary operator is applied as-expected.

**Review by alextheimer — DISMISSED** [review]
Approved -- please see the comments for a couple questions/test-suggestions.

File: prometheus/src/main/scala/filodb/prometheus/ast/Expressions.scala [hunk 2]
```diff
@@ -9,9 +9,19 @@ case class UnaryExpression(operator: Operator, operand: Expression) extends Expr
...
       throw new IllegalArgumentException(s"operator=$operator is not allowed in expression=$operand")
     }
+    convertUnaryExpression().toSeriesPlan(timeParams)
+  }
+
+  private def convertUnaryExpression() : BinaryExpression = {
     // use binary expression to implement the unary operators.
     // eg. -foo is implemented through (0 - foo).
-    BinaryExpression(Scalar(0), operator, None, operand).toSeriesPlan(timeParams)
+    operand match {
```
[the remark this answers is not in the document]
Comment (yu-shipit) [reply prometheus/src/main/scala/filodb/prometheus/ast/Expressions.scala:18]: added new tests.
unary signs are either `+` or `-`

**Review by alextheimer — APPROVED** [review]
Thanks for the updates @yu-shipit 🙏

**Pull request merged**